engine/hud/AbstractHudElement.hpp

```diff
@@ -0,0 +1,136 @@
+#ifndef ABSTRACTHUDElEMENT_HPP
+#define ABSTRACTHUDElEMENT_HPP
+
+#include <SDL2/SDL.h>
+
+#include "engine.hpp"
+
+class Camera;
+extern SDL_Renderer *_renderer;
+SDL_Texture* get_sprite(std::string);
+
+class AbstractHudElement {
+public:
+    /**
+     * If set to false, onInput will not be called by scene.
+     */
+    bool mIsInteractive = false;
+    
+    /**
+     * Base class constructor - this should be called by all children classes
+     * xPos and yPos given the position of the hud element in pixels - in what
+     * sense exactly depends on the align flags.
+     * xDim and yDim give the dimensions of the bounding box of the element in
+     * pixels.
+     * xAlignFlag and yAlignFlag determine the alignment of the bounding box.
+     * If xAlignFlag is set to -1 the bounding box will be positioned with
+     * its left side xPos pixels away from the left side of the window. If
+     * its set to 1 it will be positioned with its right side xPos pixels away
+     * from the right side of the window. Setting it to 0 will centre align the
+     * box in the x-axis (regardless of the position values). yAlignFlag works
+     * similarly with -1 corresponding to alignment relative to the top of the
+     * window and 1 corresponding to the bottom.
+     */
+    AbstractHudElement(int xPos, int yPos, int xDim, int yDim,
```

Comment:
xAlignFlag and yAlignFlags could use enums such as "LEFT", "CENTERED" and "RIGHT" for better code clarity.